lib/Transport.js

```diff
@@ -261,6 +264,12 @@ class Transport {
           return onConnectionError(
             new RequestAbortedError(`The content length (${contentLength}) is bigger than the maximum allowed string (${MAX_STRING_LENGTH})`, result)
           )
+        } else if (shouldApplyCircuitBreaker(contentLength)) {
```

Comment:
Could this be `this.memoryCircuitBreakerEnabled && shouldApplyCircuitBreaker(contentLength)`?

I would like to avoid `process.memoryUsage()` being called if this is disabled. Based on the [docs](https://nodejs.org/dist/latest-v14.x/docs/api/process.html#process_process_memoryusage):

> The process.memoryUsage() method iterates over each page to gather information about memory usage which might be slow depending on the program memory allocations.

So it seems to potentially expensive operation if used on every request.

Reply:
I may keep the else if statement simple and  do fail fast check first in the helper function like : 
```
shouldApplyCircuitBreaker(contentLength) => {
     if (! this.memoryCircuitBreakerEnabled) return false
     else ....
}
```
Do you agree on this?

Reply:
\m/ Yeah that's cool too!